catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/content/brick/WhenStartedBrickTest.java

```diff
@@ -0,0 +1,121 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.uiespresso.content.brick;
+
+import android.support.test.runner.AndroidJUnit4;
+
+import org.catrobat.catroid.R;
+import org.catrobat.catroid.content.bricks.WaitBrick;
+import org.catrobat.catroid.content.bricks.WhenStartedBrick;
+import org.catrobat.catroid.ui.ScriptActivity;
+import org.catrobat.catroid.uiespresso.content.brick.utils.BrickTestUtils;
+import org.catrobat.catroid.uiespresso.util.BaseActivityInstrumentationRule;
+import org.catrobat.catroid.uiespresso.util.UiTestUtils;
+import org.catrobat.catroid.uiespresso.util.matchers.BrickCategoryListMatchers;
+import org.catrobat.catroid.uiespresso.util.matchers.BrickPrototypeListMatchers;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import static android.support.test.espresso.Espresso.onData;
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.assertion.ViewAssertions.doesNotExist;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+import static android.support.test.espresso.matcher.ViewMatchers.withText;
+
+import static org.catrobat.catroid.uiespresso.content.brick.utils.BrickTestUtils.checkIfBrickAtPositionShowsString;
+import static org.catrobat.catroid.uiespresso.content.brick.utils.BrickTestUtils.onScriptList;
+import static org.hamcrest.Matchers.allOf;
+import static org.hamcrest.Matchers.instanceOf;
+import static org.hamcrest.core.Is.is;
+
+@RunWith(AndroidJUnit4.class)
+public class WhenStartedBrickTest {
+	@Rule
+	public BaseActivityInstrumentationRule<ScriptActivity> baseActivityTestRule = new
+			BaseActivityInstrumentationRule<>(ScriptActivity.class, true, false);
+
+	@Before
+	public void setUp() throws Exception {
+
+		BrickTestUtils.createProjectAndGetStartScript("WhenStartedBrickTest");
+		baseActivityTestRule.launchActivity(null);
+	}
+
+	@Test
+	public void whenStartedBrick() {
+
+		checkIfBrickAtPositionShowsString(0, R.string.brick_when_started);
+		deleteStartBrick(0);
+		onView(withId(R.string.brick_when_started)).check(doesNotExist());
+
+		checkChildBrickAddsWhenBrick();
+		checkTwoWhenBricksInSciptlist();
+	}
+
+	public void checkTwoWhenBricksInSciptlist() {
+		addBrickAtPosition(WaitBrick.class, 1, R.string.category_control);
+		addBrickAtPosition(WhenStartedBrick.class, 1, R.string.category_event);
+		checkIfBrickAtPositionShowsString(1, R.string.brick_when_started);
+		checkIfBrickAtPositionShowsString(0, R.string.brick_when_started);
+
+		deleteStartBrick(0);
```

Comment:
parameter 0 is unnecessary. the function is in this class, the whenstarted brick is always going to be at position 0